code/modules/atmospherics/machinery/atmosmachinery.dm

```diff
@@ -198,9 +198,14 @@
  * * direction - the direction we are checking against
  * * prompted_layer - the piping_layer we are inside
  */
-/obj/machinery/atmospherics/proc/findConnecting(direction, prompted_layer)
+/obj/machinery/atmospherics/proc/findConnecting(direction, prompted_layer, connect_all = FALSE, color_piping)
 	for(var/obj/machinery/atmospherics/target in get_step(src, direction))
 		if(target.initialize_directions & get_dir(target,src))
+			if(connect_all)
+				if(target.pipe_color != GLOB.pipe_paint_colors[color_piping] || target.piping_layer != prompted_layer)
```

Comment:
why does connect all exist in the first place? shouldn't this logic be a part of connection_check and it's subprocs?

Reply:
ah it was a remnant of old code while i was working, i'll remove it